sliding_window_max/sliding_window_max.py

```diff
@@ -2,14 +2,38 @@
 Input: a List of integers as well as an integer `k` representing the size of the sliding window
 Returns: a List of integers
 '''
-def sliding_window_max(nums, k):
+#Recursive
+def sliding_window_max(nums, k, cache={}):
     # Your code here
+    index = k - 1
 
-    pass
+    if len(nums) == k:
+        nums.sort()
+        return [nums[index]]
+    elif k in cache:
+        return cache[k]
+    else:        
+        temp = nums[:k]
+        temp.sort()
+        x = [temp[index]] + sliding_window_max(nums[1:], k, cache)
+        cache[k] = x
+        return x
```

Comment:
Overall I would say sliding window is the most useful technique out of all of these to know for your white-boarding challenges! Great job here!